cpp/oneapi/dal/algo/kmeans/train_types.cpp

```diff
@@ -107,15 +107,15 @@ void train_result<Task>::set_labels_impl(const table& value) {
 template <typename Task>
 void train_result<Task>::set_iteration_count_impl(std::int64_t value) {
     if (value < 0) {
```

Comment:
What if user set negative value with assumption that the algorithm should be stopped only by exceeding tolerance?

Reply:
This is a result of training, this field is assumed to contain actual number of iteration performed by kmeans, not requested by the user.

I guess in case of `max_iteration_count = 0` in descriptor, we do not check stop condition with respect to iteration count. Don't we @amgrigoriev?